core/kernel/versioning/class.Repository.php

```diff
@@ -400,23 +394,30 @@ public function getRootFile()
      * @author Jerome Bogaerts, <jerome@taotesting.com>
      * @param  string filePath The path to the file you want the repository to deal with.
      * @param  string label A label for the created file Resource.
+     * @param  callable $createFileName A function that generates unique file name during spawn file
      * @return core_kernel_versioning_File
      * @since 2.4
      */
-    public function spawnFile($filePath, $label = '')
+    public function spawnFile($filePath, $label = '', callable $createFileName = null)
```

Comment:
Default method hardcoded to use crc32 function, but webservices in taoQtiTest use md5. I think it's good to have ability to customise this behavior.